movement/utils/dimensions.py

```diff
@@ -0,0 +1,113 @@
+"""Utilities for manipulating dimensions of ``xarray.DataArray`` objects."""
+
+from collections.abc import Hashable, Iterable
+
+import xarray as xr
+
+
+def collapse_extra_dimensions(
+    da: xr.DataArray,
+    preserve_dims: Iterable[str] = ("time", "space"),
+    **selection: int | str,
+) -> xr.DataArray:
+    """Collapse a ``DataArray``, preserving only the specified dimensions.
+
+    By default, dimensions that are collapsed retain the corresponding 'slice'
+    along their 0th index of those dimensions, unless a particular index for is
+    given in the ``selection``.
+
+    Parameters
+    ----------
+    da : xarray.DataArray
+        DataArray of multiple dimensions, which is to be collapsed.
+    preserve_dims : Iterable[str]
+        The dimensions of ``da`` that should not be collapsed.
+    selection : int | str
+        Mapping from dimension names to a particular index in that dimension.
+        Dimensions that appear with an index in ``selection`` retain that index
+        slice when collapsed, rather than the default 0th index slice.
+
+    Returns
+    -------
+    xarray.DataArray
+        DataArray whose shape is the same as the shape of the preserved
+        dimensions of ``da``, containing the data obtained from a slice along
+        the collapsed dimensions.
+
+    Examples
+    --------
+    Collapse a ``DataArray`` down to just a ``"time"``-``"space"`` slice.
+
+    >>> import xarray as xr
+    >>> import numpy as np
+    >>> shape = (7, 2, 3, 4)
+    >>> da = xr.DataArray(
+    ...     data=np.arange(np.prod(shape)).reshape(shape),
+    ...     dims=["time", "space", "dim_to_collapse_0", "dim_to_collapse_1"],
+    ... )
+    >>> space_time = collapse_extra_dimensions(da)
+    >>> print(space_time.shape)
+    (7, 2)
+
+    The call to ``collapse_extra_dimensions`` above is equivalent to
+    ``da.sel(dim_to_collapse_0=0, dim_to_collapse_1=0)``. We can change which
+    slice we take from the collapsed dimensions by passing them as keyword
+    arguments.
+
+    >>> # Equivalent to da.sel(dim_to_collapse_0=2, dim_to_collapse_1=1)
```

Comment:
Added comments about sel and isel for clarification, like in the above comment.

```rst
    >>> # Equivalent to da.sel(keypoints="abdomen", individuals="Bob")
    >>> # or to da.isel(keypoints=2, individuals=1)
    >>> space_time_different_slice = collapse_extra_dimensions(
    ...     da, keypoints="abdomen", individuals="Bob"
    ... )
````

In the example I tend to favour label-based indexing as opposed to integer-based, because it's less error-prone imo.